### PR TITLE
✨ PreactBaseElement['staticProps']

### PR DESCRIPTION
**Pull request by alanorozco**
Related to #30280

### CODE REVIEW

**Comment by dvoytenko** [comment]
@alanorozco could you pls rebase?

**Comment by alanorozco** [comment]
@dvoytenko Done, thanks for bringing it up.

File: src/preact/base-element.js [hunk 1]
```diff
@@ -194,6 +194,11 @@ export class PreactBaseElement extends AMP.BaseElement {
       ...templatesInit,
     });
 
+    const staticProps = Ctor['staticProps'];
+    if (staticProps) {
+      Object.assign(/** @type {!Object} */ (this.defaultProps_), staticProps);
+    }
```
Comment (alanorozco) [inline src/preact/base-element.js:200]: I added this condition for consistency with below, but I don't think it's needed in either case?

image

Comment (dvoytenko) [reply src/preact/base-element.js:200]: Yeah. It's not necessary. We can even combine them into one:

```
Object.assign(this.defaultProps_, staticProps, iniProps);
```

**Review by dvoytenko — APPROVED** [review]

**Pull request merged**